draft-ietf-taps-interface.md

```diff
@@ -503,44 +503,49 @@ implemented over UDP.
 This property specifies whether the application needs to use a transport
 protocol that ensures that all data is received on the other side without
 corruption. This also entails being notified when a Connection is closed or
-aborted. The recommended default is to enable Reliable Data Transfer.
+aborted. The recommended default is to Require Reliable Data Transfer.
+
+### Preservation of Message Boundaries {#prop-boundaries}
+
+This property specifies whether the application needs to use a transport
+protocol that preserves message boundaries. The recommended default
+is to Prefer Preservation of Message Boundaries.
 
 ### Configure per-Message reliability {#prop-partially-reliable}
 
 This property specifies whether an application considers it useful to indicate
 its reliability requirements on a per-Message basis. This property applies to
-Connections and Connection Groups. The recommended default is to not have this
+Connections and Connection Groups. The recommended default is to Prefer this
 option.
 
 ### Preservation of data ordering {#prop-ordering}
 
 This property specifies whether the application wishes to use a transport
 protocol that can ensure that data is received by the application on the other
-end in the same order as it was sent. The recommended default is to preserve
-data ordering.
+end in the same order as it was sent. The recommended default is to Require
+Preservation of data ordering.
 
 ### Use 0-RTT session establishment with an idempotent Message {#prop-0rtt}
 
 This property specifies whether an application would like to supply a Message to
 the transport protocol before Connection establishment, which will then be
 reliably transferred to the other side before or during Connection
 establishment, potentially multiple times. See also {{msg-idempotent}}. The
-recommended default is to not have this option.
+recommended default is to Prefer this option.
 
 ### Multistream Connections in Group {#prop-multistream}
 
 This property specifies that the application would prefer multiple Connections
 within a Connection Group to be provided by streams of a single underlying
-transport connection where possible. The recommended default is to not have
+transport connection where possible. The recommended default is to Prefer have
 this option.
 
 ### Control checksum coverage on sending or receiving {#prop-checksum-control}
 
 This property specifies whether the application considers it useful to enable,
 disable, or configure a checksum when sending a Message, or configure whether to
-require a checksum or not when receiving.  The recommended default is full
-checksum coverage without the option to configure it, and requiring a checksum
-when receiving.
+require a checksum or not when receiving.  The recommended default is to Prefer
+this option for sending, and Require a checksum when receiving.
```

Comment:
This language was IMO not correct in the initial version; the fix to change preference is also not right... I think this is two separate properties: require checksum on receive, control checksum coverage on send.

Prefer checksum coverage control on sending will tend to always select UDP-Lite when e.g. TAPS-enable Linux machines are talking to each other. Is this what we want?

Reply:
I agree that these should be two properties. Will fix.

About always selecting UDP-Lite: why not? The problem is reachability, and that's for the TAPS transport system to figure out. It's certainly correct to use UDP - not much bad will happen.

Reply:
We already have "Ignore", which could be slightly redefined to be used for this purpose.

Reply:
Done